File: LinkedList/LinkedList.cpp
#include "LinkedList.h"
#include <iostream>
using namespace std;

template <class T>
LinkedList<T>::LinkedList() {
	sizeVar = 0;
	head = NULL;
}

template <class T>
void LinkedList<T>::add(T element, int pos) {
	if (pos < 0 || pos > sizeVar) {
		return;
	}
	else {
		if (pos == 0) {
			Node<T>* rest = head;
			head = new Node<T>(element, rest);
		}
		else {
			Node<T>* current = head;
			while (pos > 1) {
				current = current->next;
				pos--;
			}
			if (pos == sizeVar) {
				current->next = new Node<T>(element, NULL);
			}
			else {
				Node<T>* rest = current->next;
				current->next = new Node<T>(element, rest);
			}
		}
		sizeVar++;
	}
}

template <class T>
void LinkedList<T>::remove(int pos) {
	if (pos < 0 || pos > sizeVar - 1) {
		return;
	}
	else {
		if (pos == 0) {
			head = head->next;
		}
		else {
			Node<T>* current = head;
			while (pos > 1) {
				current = current->next;
				pos--;
			}
			if (pos == sizeVar - 1) {
				current->next = NULL;
			}
			else {
				Node<T>* newNext = current->next->next;
				current->next = newNext;
			}
		}
		sizeVar--;
	}
}

template <class T>
T LinkedList<T>::get(int pos) {
	if (pos < 0 || pos > sizeVar - 1) {
		return 0;
	}
	Node<T>* current = head;
	while (pos > 0) {
		current = current->next;
		pos--;
	}
	return current->element;
}

template <class T>
void LinkedList<T>::set(int pos, T value) {
	if (pos < 0 || pos > sizeVar - 1) {
		return;
	}
	Node<T>* current = head;
	while (pos > 0) {
		current = current->next;
		pos--;
	}
	current->element = value;
}

template <class T>
int LinkedList<T>::size() {
	return sizeVar;
}

int main() {
	
}

File: LinkedList/LinkedList.go
package main

import "fmt"

type Node struct {
	element int
	next *Node
}

type LinkedList struct {
	sizeVar int
	head *Node
}

func (list *LinkedList) Add(value int, pos int) {
	if (pos < 0 || pos > list.sizeVar) {
		return
	} else {
		if (pos == 0) {
			rest := list.head
			list.head = &Node{value, rest}
		} else {
			current := list.head
			for ; pos > 1 ; {
				current = current.next
				pos--
			}
			if (pos == list.sizeVar) {
				current.next = &Node{value, nil}
			} else {
				rest := current.next
				current.next = &Node{value, rest}
			}
		}
		list.sizeVar++
	}
}

func (list *LinkedList) Remove(pos int) {
	if (pos < 0 || pos > list.sizeVar - 1) {
		return
	} else {
		if (pos == 0) {
			list.head = list.head.next
		} else {
			current := list.head
			for ; pos > 1 ; {
				current = current.next
				pos--
			}
			if (pos == list.sizeVar - 1) {
				current.next = nil
			} else {
				newNext := current.next.next
				current.next = newNext
			}
		}
		list.sizeVar--
	}
}

func (list *LinkedList) Get(pos int) int {
	if (pos < 0 || pos >= list.sizeVar) {
		return 0
	}

	current := list.head
	for ; pos > 0 ; {
		current = current.next
		pos--
	}
	return current.element
}

func (list *LinkedList) Set(pos int, value int) {
	if (pos < 0 || pos >= list.sizeVar) {
		return
	}
	current := list.head
	for ; pos > 0 ; {
		current = current.next
		pos--
	}
	current.element = value
}

func (list *LinkedList) Size() int {
	return list.sizeVar
}

func main () {
	myList := LinkedList{0, nil}
	myList.Add(1, 0)
	fmt.Println(myList.Get(0))
}